### PR TITLE
Move sign up and login form to modal that overlays on page with map

### LINKED ISSUE #44: Login / sign in redirect to index
Submitting login or signin forms should send the user back to the index page.

### PR DESCRIPTION
**Pull request by PriyaVarra**
This pull request transports the html for the signup form in signup.html to inside a modal in index.html and the html for the login form in login.html to inside a modal in index.html. This means signup.html and login.html should be deleted once this request is merged with master. It also adds some functions in signup.js  and login.js to clear all the text fields in the form once the modal is hidden and hide the modal without refreshing the page if the user successfully signs up or logs in.

This should be compatible with any changes in the validation scheme that may continued to be added to signup.js or login.js. Any changes to the signup or login forms will now have to be done in index.html.

### CODE REVIEW

**Comment by PriyaVarra** [comment]
@andy7809 @emmapaczkowski Before merging this pull request, should I also do the same for the form in login.html?

File: where-to-travel/src/main/webapp/index.html [hunk 1]
```diff
@@ -122,6 +123,36 @@ <h5> Finding Attractions... </h5>
...
+                    <input type="password" class="form-control" minlength="6" id="passwordSignUp" oninvalid="this.setCustomValidity('Password must be at least 6 characters.')">
+                </div>
+                <div class="form-group">
+                    <label for="passwordConfirmation" class="bmd-label-floating">Confirm password</label>
+                    <input type="password" class="form-control" id="passwordConfirmation">
+                </div>
+                <div class="text-center">
+                    <button type="submit" id="submit" class="btn btn-primary">Sign Up</button>
+                    <button type="button" id="back"  class="btn btn-primary" data-dismiss="modal">Cancel</button>
+                </div>
```
Comment (emmapaczkowski) [inline where-to-travel/src/main/webapp/index.html]: Is this 2 spaces or 4? It looks like 4 but I could be wrong.

Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/index.html]: You're right. I have fixed this in the latest commits.

**Review by andy7809 — COMMENTED** [review]
Not too worried about signup.js or login.js, I cleaned those up in my PR and will add some more changes later.

On another note, your frontend/material design stuff looks great! Good job!!!

**Comment by andy7809** [comment]
LGTM, looks ready to go

File: where-to-travel/src/main/webapp/login.js [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
-  });
-  promise.catch(e => {
-    console.log(e.message);
-    alert(e.message);
+// Preferred method of material form behavior interaction is jQuery
+// When the document is loaded, add validity check to form on submit.
+$(document).ready(function() {
+  $(document).on('submit', '#loginForm', function() {
+    const email = emailLogin.value;
+    const pass = passwordLogin.value;
```
Comment (wjiamin) [inline where-to-travel/src/main/webapp/login.js]: nit: prefer not shortening names, ->password?

**Review by wjiamin — COMMENTED** [review]
LGTM! I'll let Sailesh approve.

File: where-to-travel/src/main/webapp/login.js (hunk 2 above)
Comment (PriyaVarra) [reply where-to-travel/src/main/webapp/login.js]: fixed!

**Review by saileshsimhadri — APPROVED** [review]

**Pull request merged**